### DISCUSSION TITLE
Keycloak in docker compose: how to depends_on until it is completely initialised?

### DISCUSSION THREAD

**Discussion by flowt-au** [keycloak/keycloak#17637] [Q&A]
I have a docker compose file with a number of containers including Keycloak 21.0.1 and my Quarkus app. The Quarkus app uses Keycloak as an OIDC provider so Keycloak must be up and running before the Quarkus app starts (because the quarkus-oidc OidcCommonUtils) tries to connect to Keycloak as part of the standard Quarkus OIDC startup in my app.

I need to wait for `Keycloak 21.0.1 on JVM (powered by Quarkus 2.13.7.Final) started in 5.417s. Listening on: http://0.0.0.0:8080` to happen before docker compose will start my Quarkus app.

I tried `depends_on` which "works" except it resolves as soon as Keycloak starts, seconds before the `Listening` stage happens.

Any suggestions?
Thanks,
Murray

**Comment by slominskir** [comment]
If you setup a health check you can wait on healthy condition:  https://stackoverflow.com/a/74449969/562644

Another approach is to set a custom entrypoint script that polls a health check such as:  https://github.com/slominskir/keycloak_ldap/blob/8cc9baf18d082f0ef79594426eea4f45ab198739/keycloak/init/entrypoint.sh#L13-L17
